torchmetrics/utilities/checks.py

```diff
@@ -24,37 +25,425 @@ def _check_same_shape(pred: torch.Tensor, target: torch.Tensor):
         raise RuntimeError("Predictions and targets are expected to have the same shape")
 
 
+def _basic_input_validation(preds: torch.Tensor, target: torch.Tensor, threshold: float, is_multiclass: bool):
+    """
+    Perform basic validation of inputs that does not require deducing any information
+    of the type of inputs.
+    """
+
+    if target.is_floating_point():
+        raise ValueError("The `target` has to be an integer tensor.")
+    if target.min() < 0:
+        raise ValueError("The `target` has to be a non-negative tensor.")
+
+    preds_float = preds.is_floating_point()
+    if not preds_float and preds.min() < 0:
+        raise ValueError("If `preds` are integers, they have to be non-negative.")
+
+    if not preds.shape[0] == target.shape[0]:
+        raise ValueError("The `preds` and `target` should have the same first dimension.")
+
+    if preds_float and (preds.min() < 0 or preds.max() > 1):
+        raise ValueError("The `preds` should be probabilities, but values were detected outside of [0,1] range.")
+
+    if not 0 < threshold < 1:
```

Comment:
```suggestion
    if not 0 <= threshold <= 1:
```
@Borda Could it be any of these?

Reply:
Practically, it wouldn't make sense tho.